plugins/by-name/parrot/default.nix

```diff
@@ -0,0 +1,32 @@
+{ lib, ... }:
+lib.nixvim.plugins.mkNeovimPlugin {
+  name = "parrot";
+  packPathName = "parrot.nvim";
+  package = "parrot-nvim";
+
+  maintainers = [ lib.maintainers.GaetanLepage ];
+
+  extraConfig = {
+    dependencies.ripgrep.enable = lib.mkDefault true;
+  };
+
+  settingsExample = {
+    cmd_prefix = "Parrot";
+    providers = {
+      github = {
+        api_key.__raw = "os.getenv 'GITHUB_TOKEN'";
+        topic.model = "gpt-4o";
+      };
+    };
+    hooks = {
+      Ask.__raw = ''
+        function(parrot, params)
+          local template = "Please, answer to this question: {{command}}."
+          local model_obj = parrot.get_model("command")
+          parrot.logger.info("Asking model: " .. model_obj.name)
+          parrot.Prompt(params, parrot.ui.Target.popup, model_obj, "🤖 Ask ~ ", template)
+        end
+      '';
```

Comment:
FYI this will render horribly in the docs.

`renderOptionValue` uses `generators.toPretty` for non-literal values, which uses `toJSON` for string values.

Therefore you'll always get a single line string with `\n` for newlines.

I'd recommend using literalExpression when you want to manually enforce a specific formatting; in this case we want to enforce using an indented string.